sdk/identity/src/token_credentials/default_credentials.rs

```diff
@@ -134,16 +136,17 @@ impl TokenCredential for DefaultCredential {
     type Error = DefaultCredentialError;
     /// Try to fetch a token using each of the credential sources until one succeeds
     async fn get_token(&self, resource: &str) -> Result<TokenResponse, Self::Error> {
+        let mut errors = Vec::new();
         for source in &self.sources {
             let token_res = source.get_token(resource).await;
 
             if let Ok(token) = token_res {
                 return Ok(token);
             } else {
-                debug!("Failed to get credentials: {:?}", token_res.err().unwrap());
+                errors.push(token_res.err().unwrap());
```

Comment:
I know that this was how we were doing things previously, but instead of using `err().unwrap()` we should just `match` on `token_res` so that we don't need to unwrap